src/tools/run-make-support/src/lib.rs

```diff
@@ -295,6 +295,26 @@ pub fn not_contains<P: AsRef<Path>>(path: P, expected: &str) -> bool {
     !path.as_ref().file_name().is_some_and(|name| name.to_str().unwrap().contains(expected))
 }
 
+/// Builds a static lib (`.lib` on Windows MSVC and `.a` for the rest) with the given name.
+#[track_caller]
+pub fn build_native_static_lib(lib_name: &str) -> PathBuf {
+    let obj_file = if is_msvc() { format!("{lib_name}") } else { format!("{lib_name}.o") };
```

Comment:
Question: should this be a helper?

Reply:
If this becomes a helper, then `cc` should receive a rework. I can't really name this `object_file_name` or such, because it provides no extension for the MSVC platform and `.o` for others. The reason for this isn't that MSVC object files have no extension - it's that `cc` automatically appends `.obj` to output filenames. Once `cc` is done, then I need to manually add `.obj` to the filename, or else `llvm_ar` will complain, as that one *doesn't* automatically append extensions.

There is some inconsistency like this around the run_make_support library. For example, `rustc` will not auto-append `.rs` to input files, but `run` will auto-append `.exe` to input files if on Windows.

I feel like doing this rework (which would change a lot of test files too) would be better suited to a different PR.

Reply:
Good point, then let's keep it as-is for now.